zkevm-circuits/src/tx_circuit.rs

```diff
@@ -1690,6 +1715,138 @@ impl<F: Field> SubCircuitConfig<F> for TxCircuitConfig<F> {
+                is_hash_length(meta),
+            ]);
+
+            vec![
+                1.expr(),                                           // q_enable
+                meta.query_advice(tx_table.value, Rotation::cur()), // exponent
+                meta.query_advice(pow_of_rand, Rotation::cur()),    // pow_of_rand
+            ]
+            .into_iter()
+            .zip(pow_of_rand_table.table_exprs(meta))
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+
+        meta.create_gate("One chunk_txbytes_len_acc, chunk_txbytes_rlc value and pow_of_rand for each tx (in fixed section)", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            // chunk_txbytes_len_acc, chunk_txbytes_rlc and pow_of_rand stay the same for the same tx
+            cb.require_equal(
+                "chunk_txbytes_len_acc' == chunk_txbytes_len_acc",
+                meta.query_advice(chunk_txbytes_len_acc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_len_acc, Rotation::prev()),
+            );
+            cb.require_equal(
+                "chunk_txbytes_rlc' == chunk_txbytes_rlc",
+                meta.query_advice(chunk_txbytes_rlc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_rlc, Rotation::prev()),
+            );
+            cb.require_equal(
+                "pow_of_rand' == pow_of_rand",
+                meta.query_advice(pow_of_rand, Rotation::cur()),
+                meta.query_advice(pow_of_rand, Rotation::prev()),
+            );
+
+            cb.gate(and::expr([
+                meta.query_fixed(q_enable, Rotation::cur()),
+                not::expr(meta.query_fixed(q_first, Rotation::cur())),
+                not::expr(is_nonce(meta)),
+                not::expr(meta.query_advice(is_calldata, Rotation::cur()))
+            ]))
+        });
+
+        meta.create_gate(
+            "One chunk_txbytes_hash for the chunk within the fixed section",
+            |meta| {
+                let mut cb = BaseConstraintBuilder::default();
+
+                // chunk_txbytes_hash stays the same for the entire chunk
+                cb.require_equal(
+                    "chunk_txbytes_hash' == chunk_txbytes_hash",
+                    meta.query_advice(chunk_txbytes_hash, Rotation::cur()),
+                    meta.query_advice(chunk_txbytes_hash, Rotation::prev()),
+                );
+
+                cb.gate(and::expr([
+                    meta.query_fixed(q_enable, Rotation::cur()),
+                    not::expr(meta.query_fixed(q_first, Rotation::cur())),
+                    not::expr(meta.query_fixed(q_first, Rotation::prev())),
+                    not::expr(meta.query_advice(is_calldata, Rotation::cur())),
+                    not::expr(meta.query_advice(is_access_list, Rotation::cur())),
+                ]))
+            },
+        );
+
+        meta.create_gate(
```

Comment:
We are never querying these fields in the dynamic sections (access list and calldata) right? So then isn't this over-constraining?

Reply:
Removed.